src/server/replication_buffer.rs

```diff
@@ -1,377 +0,0 @@
-    pub(super) fn write_replication_id(
-        &mut self,
-        replication_id: ReplicationId,
-    ) -> bincode::Result<()> {
-        if self.entity_data_len == 0 {
-            self.write_data_entity()?;
-        }
-
-        DefaultOptions::new().serialize_into(&mut self.cursor, &replication_id)?;
-        self.entity_data_len += 1;
-
-        Ok(())
-    }
-
-    /// Removes entity data elements from `other` and copies it.
-    ///
-    /// Ends entity data for `other`.
-    /// See also [`Self::start_entity_data`] and [`Self::end_entity_data`].
-    pub(super) fn take_entity_data(&mut self, other: &mut Self) {
-        if other.entity_data_len != 0 {
-            let slice = other.as_slice();
-            let offset =
-                other.entity_data_len_pos as usize + mem::size_of_val(&other.entity_data_len);
-            self.cursor.write_all(&slice[offset..]).unwrap();
-            self.entity_data_len += other.entity_data_len;
-
-            other.entity_data_len = 0;
-        }
-
-        other.cursor.set_position(other.entity_data_pos);
-    }
-
-    /// Crops empty arrays at the end.
-    ///
-    /// Should only be called after all arrays have been written, because
-    /// arrays removed somewhere the middle cannot be detected during deserialization.
-    pub(super) fn trim_empty_arrays(&mut self) {
-        debug_assert!(!self.inside_array);
-        debug_assert_eq!(self.array_len, 0);
-        debug_assert_eq!(self.entity_data_len, 0);
-
-        let extra_len = self.trailing_empty_arrays * mem::size_of_val(&self.array_len);
-        self.cursor
-            .set_position(self.cursor.position() - extra_len as u64);
-    }
-}
-
-impl Default for ReplicationBuffer {
-    fn default() -> Self {
-        Self {
-            cursor: Default::default(),
-            array_pos: Default::default(),
-            array_len: Default::default(),
-            inside_array: Default::default(),
-            arrays_with_data: Default::default(),
-            trailing_empty_arrays: Default::default(),
-            entity_data_pos: Default::default(),
-            entity_data_len_pos: Default::default(),
-            entity_data_len: Default::default(),
-            data_entity: Entity::PLACEHOLDER,
-        }
-    }
-}
-
-/// Serializes `entity` by writing its index and generation as separate varints.
-///
-/// The index is first prepended with a bit flag to indicate if the generation
-/// is serialized or not (it is not serialized if equal to zero).
-fn serialize_entity(cursor: &mut Cursor<Vec<u8>>, entity: Entity) -> bincode::Result<()> {
-    let mut flagged_index = (entity.index() as u64) << 1;
-    let flag = entity.generation() > 0;
-    flagged_index |= flag as u64;
-
-    cursor.write_u64_varint(flagged_index)?;
-    if flag {
-        cursor.write_u32_varint(entity.generation())?;
-    }
-
-    Ok(())
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn trimming_arrays() -> bincode::Result<()> {
```

Comment:
It's intentional. Instead of modifying the message after calling `send`, I just do cut them when I do slicing. See `as_slice` for `InitMessage`, it's different from `UpdateMessage`.